From 028529e33ed45507bcb1f3eb2722de3344eea091 Mon Sep 17 00:00:00 2001
From: Thomas Haller <thaller@redhat.com>
Date: Mon, 5 Feb 2024 13:09:02 +0100
Subject: [PATCH 24/26] v2.2.0: improvement(policy): extract helper function
 for writing limit rule element

Soon the Rich_Limit will also get a burst attribute. Then _handler_add_rich_limit()
will become more complicated. We wouldn't want to duplicated that code.

(cherry picked from commit f662606891569f09553c73023a2f70086d137512)
---
 src/firewall/core/io/policy.py | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/src/firewall/core/io/policy.py b/src/firewall/core/io/policy.py
index 514a20251ef4..66535e0d0368 100644
--- a/src/firewall/core/io/policy.py
+++ b/src/firewall/core/io/policy.py
@@ -372,6 +372,11 @@ def common_check_config(obj, config, item, all_config):
                         ),
                     )
 
+
+def _handler_add_rich_limit(handler, limit):
+    handler.simpleElement("limit", {"value": limit.value})
+
+
 def common_writer(obj, handler):
     # short
     if obj.short and obj.short != "":
@@ -533,8 +538,7 @@ def common_writer(obj, handler):
                 handler.ignorableWhitespace("    ")
                 handler.startElement("log", attrs)
                 handler.ignorableWhitespace("\n      ")
-                handler.simpleElement("limit",
-                                      { "value": rule.log.limit.value })
+                _handler_add_rich_limit(handler, rule.log.limit)
                 handler.ignorableWhitespace("\n    ")
                 handler.endElement("log")
             else:
@@ -549,8 +553,7 @@ def common_writer(obj, handler):
                 handler.ignorableWhitespace("    ")
                 handler.startElement("audit", { })
                 handler.ignorableWhitespace("\n      ")
-                handler.simpleElement("limit",
-                                      { "value": rule.audit.limit.value })
+                _handler_add_rich_limit(handler, rule.audit.limit)
                 handler.ignorableWhitespace("\n    ")
                 handler.endElement("audit")
             else:
@@ -579,8 +582,7 @@ def common_writer(obj, handler):
                 handler.ignorableWhitespace("    ")
                 handler.startElement(action, attrs)
                 handler.ignorableWhitespace("\n      ")
-                handler.simpleElement("limit",
-                                      { "value": rule.action.limit.value })
+                _handler_add_rich_limit(handler, rule.action.limit)
                 handler.ignorableWhitespace("\n    ")
                 handler.endElement(action)
             else:
-- 
2.43.0

